db/migrate/20151120153254_delete_bootable_interface.rb

```diff
@@ -0,0 +1,11 @@
+class DeleteBootableInterface < ActiveRecord::Migration
+  def up
+    Nic::Base.where(:type => 'Nic::Bootable').each do |nic|
```

Comment:
Iterating/instantiating these models won't work as the class no longer exists:

<pre>
==  DeleteBootableInterface: migrating ========================================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

The single-table inheritance mechanism failed to locate the subclass: 'Nic::Bootable'. This error is raised because the column 'type' is reserved for storing the class in case of inheritance. Please rename this column if you didn't intend it to be used for storing the inheritance class or overwrite Nic::Base.inheritance_column to use another column for that information./home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/inheritance.rb:143:in `rescue in find_sti_class'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/inheritance.rb:136:in `find_sti_class'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/inheritance.rb:62:in `instantiate'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/querying.rb:38:in `block (2 levels) in find_by_sql'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/querying.rb:38:in `collect!'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/querying.rb:38:in `block in find_by_sql'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/explain.rb:41:in `logging_query_plan'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/querying.rb:37:in `find_by_sql'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/relation.rb:171:in `exec_queries'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/relation.rb:160:in `block in to_a'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/explain.rb:34:in `logging_query_plan'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/relation.rb:159:in `to_a'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/bullet-4.14.10/lib/bullet/active_record3x.rb:10:in `to_a'
/home/dcleal/.rvm/gems/ruby-2.0.0-p353@foreman/gems/activerecord-3.2.22/lib/active_record/relation/delegation.rb:6:in `each'
/home/dcleal/code/foreman/foreman/db/migrate/20151120153254_delete_bootable_interface.rb:3:in `up'
</pre>

This appears to work better: `Nic::Base.where(:type => 'Nic::Bootable').update_all(:type => 'Nic::Managed', :provision => true)`, but won't trigger validations.  I'm not sure how you'd do that - possibly defining a fake bootable model in this migration, then using STI "becomes" to convert it and then validate the Nic::Managed.  I shouldn't worry about that...